pkg/virt-handler/virtwrap/manager.go

```diff
@@ -61,6 +63,48 @@ type LibvirtDomainManager struct {
 	podIsolationDetector isolation.PodIsolationDetector
 }
 
+func (l *LibvirtDomainManager) CreateNetwork(netXML *string, netName string) error {
+	network, err := l.virConn.LookupNetworkByName(netName)
+	if err != nil {
+		network, err = l.virConn.NetworkDefineXML(*netXML)
+		if err != nil {
+			logging.DefaultLogger().Error().Reason(err).Msgf("defining network from xml: %s", *netXML)
+			return err
+		}
+	}
+
+	err = network.SetAutostart(true)
+	if err != nil {
+		logging.DefaultLogger().Error().Reason(err).Msgf("Setting network to autostart: %s", *netXML)
+
+		return goerrors.New(fmt.Sprintf("Cannot set network %s to autostart", netName))
+	}
+
+	active, err := network.IsActive()
+	if err != nil || !active {
+		err = network.Create()
+		if err != nil {
+			return goerrors.New(fmt.Sprintf("Network %s creation failed.", netName))
+		}
+	}
+
+	return nil
+}
+func (l *LibvirtDomainManager) DestroyNetwork(netName string) error {
+	network, err := l.virConn.LookupNetworkByName(netName)
+	if err != nil {
+		logging.DefaultLogger().Warning().Reason(err).Msgf("Failed to find network: %s", netName)
+		return nil
+	}
+	err = network.Destroy()
```

Comment:
I'm pretty sure after destroy() is called, undefine() needs to be called as well.  From what I'm reading, destroy just shuts down the network, but the definition will still be persisted by libvirt until undefine is called.

I'm not a fan for the libvirt api's choice of terms .

Reply:
I think you are right and this is not enough. However, do we need the network to be persisted across reboots? If not we could replace NetworkDefine with NetworkCreate, the the Destroy would be enough:

"net-define - An alternative to net-create. Use this when you want a persistent virtual network that will last through reboots and shutdowns, rather than a transient one created using net-create. "